File: LC_JS-SOLUTIONS/535. Encode and Decode TinyURL.js
/**
 * Encodes a URL to a shortened URL.
 *
 * @param {string} longUrl
 * @return {string}
 */

var encodeDic = {};
var decodeDic = {};
var base = "http://tinyurl.com/";

var encode = function (longUrl) {
  let shortUrl = this.base + Math.random().toString(36).substr(2, 7);
  encodeDic[longUrl] = shortUrl;
  decodeDic[shortUrl] = longUrl;
  return shortUrl;
};

/**
 * Decodes a shortened URL to its original URL.
 *
 * @param {string} shortUrl
 * @return {string}
 */
var decode = function (shortUrl) {
  return decodeDic[shortUrl];
};

/**
 * Your functions will be called as such:
 * decode(encode(url));
 */

File: LC-Python-Solution/535. Encode and Decode TinyURL.py
class Codec:

    def __init__(self):
        self.encodeDic = {}
        self.decodeDic = {}
        self.base = "http://timyurl.com/"

    def encode(self, longUrl: str) -> str:
        """Encodes a URL to a shortened URL.
        """
        if longUrl not in self.encodeDic:
            shortUrl = self.base + str(len(self.encodeDic))
            self.encodeDic[longUrl] = shortUrl
            self.decodeDic[shortUrl] = longUrl
        
        return self.encodeDic[longUrl]        

    def decode(self, shortUrl: str) -> str:
        """Decodes a shortened URL to its original URL.
        """
        return self.decodeDic[shortUrl]

# Your Codec object will be instantiated and called as such:
# codec = Codec()
# codec.decode(codec.encode(url))